kernel/src/engine/default/parquet.rs

```diff
@@ -27,6 +34,66 @@ pub struct DefaultParquetHandler<E: TaskExecutor> {
     readahead: usize,
 }
 
+/// Metadata of a parquet file, currently just includes the file metadata but will expand to
+/// include file statistics and other metadata in the future.
+#[derive(Debug)]
+pub struct ParquetMetadata {
```

Comment:
suggest giving this a different name to avoid confusion with the parquet crate's [metadata](https://docs.rs/parquet/latest/parquet/file/metadata/index.html)

I wanted to suggest just using their types, since there is overlap, but this feels more like maybe `ParquetWriteMetadata` or something, i.e. metadata relating to the write rather than the parquet file itself.

Also, all the stuff that goes into `WRITE_METADATA_SCHEMA` is buried in the `FileMeta`. I'd prefer to have a `struct`  that could `#[derive(Schema)]` for the metadata than have the schema manually declared. AFAICT you're not using the fact that this is a `FileMeta` anywhere, and are just using it as a container inside a container.

Reply:
yup - good point! originally, I just had `ParquetMetadata` with path, size, modification_time and just realized that was the same as FileMeta so tried to unify there. It sounds like instead you're thinking of just flattening those fields under a `ParquetWriteMetadata` struct that will live in the kernel (instead of just in default engine) and that uses `#[derive(Schema)]` to get the 'write metadata schema' instead of the separate schema definition.

That seems to be more elegant thanks! :)

Reply:
Actually, thinking about this more, we have two steps to this data flow:
1. engine collecting the data returned from each parquet file written:
  - path
  - size
  - last_modified
2. engine communicating to kernel the data required for each write:
  - path
  - size
  - last_modified
  - partition_values
  - data_change

In (2) we add partition_values and data_change, which wouldn't necessarily be inherent to the parquet write but layers on more information about that file (e.g. engine just wrote this parquet file corresponding to `partition_col = part1`)

I had introduced `ParquetMetadata` to house (1) which is purely an engine concept. Then, in kernel I had `WRITE_METADATA_SCHEMA` which specified the schema for (2).

Thinking more about the suggestion, maybe we need a rename of (1) and then maybe change (2) to be a struct that does `#[derive(Schema)]`? I'm thinking now about reasons for struct vs. just the schema definition like I have but wanted to post this comment to add some more context and then we can discuss more :)